### DISCUSSION TITLE
Change Prefix of Implementation class

### DISCUSSION THREAD

**Discussion by zyberzebra** [mapstruct/mapstruct#3612] [Ideas]
Now we can do this:
`@Mapper(implementationName = "MyCustomPrefix<CLASS_NAME>MyCustomSuffix")`

For purposes of illustration you have interfaces like `InterfaceCarMapper`, `InterfaceInventoryItemMapper` and would like to be generated:
`CarMapperImpl`, `InventoryItemMapper` 

Could we add the possibility to trim or replace leading or trailing characters?

I'm not sure how a good implementation of that feature could look like, but it would be really nice!

```
@Mapper(implementationName = "<CLASS_NAME>", trimLeading = "Interface")
@Mapper(implementationName = "<CLASS_NAME>", replaceSubstring = "Interface", replacement = "")
@Mapper(implementationName = "<CLASS_NAME>", trimStartIndex = 9)
@Mapper(implementationName = "(-9)<CLASS_NAME>")
@Mapper(implementationName = "(Interface)<CLASS_NAME>")
```

**Comment by filiphr** [comment]
@zyberzebra I'm not fond of adding things such as `trimLeading`, `replaceString` or rather anything else special for the `implementationName`.

However, I would be open to providing an SPI, where we will pass the configured `implementatioName` and the name of the class and then this SPI can provide us with the name that should be used in the generated code.

Having said that, having something like `InterfaceInventoryItemMapper` as your interface and `InventoryItemMapper` as the impl class is a recipe for disaster. In my auto complete I would mostly use `InventoryItemMapper`.

**Reply by zyberzebra** [reply]
> Having said that, having something like `InterfaceInventoryItemMapper` as your interface and `InventoryItemMapper` as the impl class is a recipe for disaster. In my auto complete I would mostly use `InventoryItemMapper`.

I fully agree! A lot of legacy code bases have that leading "I" as convention for interfaces though :/

**Reply by filiphr** [reply]
Legacy code can indeed be tricky. Feel free to create an issue for this

**Comment by zyberzebra** [comment]
Created an Issue/Feature Request
